### DISCUSSION TITLE
Line too Long

### DISCUSSION THREAD

**Discussion by christianbraeunlich** [microsoft/alguidelines#155] [BC Best Practices]
🦦 Started from https://github.com/StefanMaron/BusinessCentral.LinterCop/discussions/176

---

I'd like to avoid the lines of code become too long

Inspired by:
- https://peps.python.org/pep-0008/#maximum-line-length
- https://www.flake8rules.com/rules/E501.html

---

MS Docs on Line length: https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/compliance/apptest-bestpracticesforalcode#line-length

## Bad code

```al
LibraryPriceCalculation.CreatePriceHeader(PriceListHeader, "Price Type"::Sale, "Price Source Type"::"All Customers", '');
LibraryPriceCalculation.CreateSalesPriceLine(PriceListLine, PriceListHeader.Code, "Price Source Type"::"All Customers", '', "Price Asset Type"::"G/L Account", GLAccount."No.");
```

## Good code
```al
LibraryPriceCalculation.CreatePriceHeader(PriceListHeader, "Price Type"::Sale, "Price Source Type"::"All Customers", '');
LibraryPriceCalculation.CreateSalesPriceLine(
    PriceListLine, PriceListHeader.Code, "Price Source Type"::"All Customers", '', "Price Asset Type"::"G/L Account", GLAccount."No.");
```

With a maximum of `152` characters per line (the maximum length should be adjustable).

---

Maybe this comes with an exception for procedures:

```al
local procedure CreatePurchPriceLinesForJobs(Job: Array[2] of Record Job;JobTask: Record "Job Task"; var PriceListLine: Array[5] of Record "Price List Line"; FifthPriceAmountType: Enum "Price Amount Type")
```

because I don't like this:

```al
local procedure CreatePurchPriceLinesForJobs(Job: Array[2] of Record Job; JobTask: Record "Job Task";
    var PriceListLine: Array[5] of Record "Price List Line"; FifthPriceAmountType: Enum "Price Amount Type")
```

**Comment by mjmatthiesen** [comment]
I have mine at 128, because it works out with my layout so I can have two open side by side when needed.

I don't split proc headers into new lines, though I don't have a problem with it either.
I prefer indenting to the opening bracket rather than a single indent on the new line, but that's just because I find that easier to read.
